x-pack/plugins/fleet/public/hooks/use_request/settings.ts

```diff
@@ -41,6 +41,17 @@ export function sendGetSettings() {
   });
 }
 
+export function usePutSettingsMutation() {
+  const queryClient = useQueryClient();
+
+  return useMutation({
+    mutationFn: sendPutSettings,
+    onSuccess: async () => {
+      await queryClient.invalidateQueries(['settings']);
```

Comment:
this will refetch the invalid queries, I personally think it's a much better pattern than having to pass a callback through a large react tree.

Reply:
Yeah this a hugely useful feature - glad to see it put to use here!

(nit) We don't need to `await` query invalidation https://tanstack.com/query/v4/docs/react/guides/invalidations-from-mutations.